tests/adapters/base.py

```diff
@@ -30,55 +70,167 @@ def test_initialize_model_input(self, test_adapter):
         """
         Test that initalizing model input does as intended.
         """
+        set_values = [0.9, 1.1]
+        test_adapter._output.get_writable_timeseries_view(
+            ("Emissions", "CO2"),
+            ("World",),
+            "GtC/yr",
+            [-10, 0, 10],
+            ParameterType.POINT_TIMESERIES,
+        ).set(set_values)
+
         assert not test_adapter._initialized
-        # TODO test for missing but mandatory parameter
-        test_adapter.initialize_model_input()
+        # start and end time should be set here as they're part of setting the
+        # timeseries views of emissions/concentrations/other drivers
+        test_adapter.initialize_model_input(start_time, end_time)
         assert test_adapter._initialized
+        # TODO test for start time and end time being passed properly
+        assert test_adapter._output.get_writable_timeseries_view(
+            ("Emissions", "CO2"),
+            ("World",),
+            "GtC/yr",
+            [-10, 0, 10],
+            ParameterType.POINT_TIMESERIES,
+        ).get() == set_values
 
-    def test_initialize_model_input_non_model_parameter(self, test_adapter):
-        tname = ("junk",)
-        test_adapter._parameters.get_writable_scalar_view(tname, ("World",), "K").set(4)
-        test_adapter.initialize_model_input()
-        # TODO test that "junk" has not been used
+        # TODO test for missing but mandatory input
+        # We could turn this into an abstract method so that implementers are forced
+        # to do an assertion like the below.
+        test_adapter._output.get_writable_timeseries_view(
+            ("Emissions", "junk"),
+            ("World",),
+            "GtC/yr",
+            [-10, 0, 10],
+            ParameterType.POINT_TIMESERIES,
+        ).set(set_values)
+        # if you try to initialise and run with a timeseries that can't be used,
+        # you get a warning
+        assert raises_warning(test_adapter.initialize_model_input(
```

Comment:
I'd very much prefer to have a method to get used/unused parameters rather than throwing warnings

Reply:
I suspect this will end up being personal preference. I'm pretty relaxed, the reason I went for warning was that one of the most common complaints I've heard about these models is that they do things which aren't expected/don't match people's assumptions. If you throw a warning then people are forced to look at it, whereas if you have a method to get used/unused parameters then people will only work it out if they go looking. Given people could have gone looking already but didn't, it seems to me they need to have things raised in order for them to realise that things aren't working as they expected.

Reply:
Following other discussions, I agree so this warning can go elsewhere, see https://github.com/openclimatedata/openscm/issues/123#issuecomment-492926301